### PR TITLE
complete unittests for graphite adapter

### PR DESCRIPTION
**Pull request by evemorgen**

### CODE REVIEW

**Comment by evemorgen** [comment]
I think it should be 100% coverage, does some tool should highlight it?

File: rails_app/spec/adapters/graphite_adapter_spec.rb [hunk 1]
```diff
@@ -0,0 +1,126 @@
+require 'rails_helper'
+require 'webmock/rspec'
+
+RSpec.describe GraphiteAdapter do
```
Comment (szmo) [inline rails_app/spec/adapters/graphite_adapter_spec.rb:4]:
```
RSpec.describe GraphiteAdapter do
here before or let if needed, should work well also
  describe '.method_name' do
      it 'some text to sound well written as plain text' do
        expects
      end
```

File: rails_app/spec/examples.txt [hunk 2]
```diff
@@ -1,3 +1,21 @@
-example_id               | status | run_time        |
------------------------- | ------ | --------------- |
-./spec/test_spec.rb[1:1] | passed | 0.00082 seconds |
+example_id                                         | status  | run_time        |
```
Comment (szmo) [inline rails_app/spec/examples.txt:1]: pls add it to .gitignore

**Review by szmo — CHANGES_REQUESTED** [review]

File: rails_app/spec/adapters/graphite_adapter_spec.rb (hunk 1 above)
Comment (evemorgen) [reply rails_app/spec/adapters/graphite_adapter_spec.rb:4]: done

File: rails_app/spec/examples.txt (hunk 2 above)
Comment (evemorgen) [reply rails_app/spec/examples.txt:1]: done

File: rails_app/spec/adapters/graphite_adapter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      expect(@graphite).to receive(:get_metric)
+        .with('mock.metric', 'png', time_from: nil, time_to: nil)
+        .and_return(mock_values)
+
+      expect(@graphite.get_metric_png('mock.metric')).to eql(mock_values)
+    end
+  end
+
+  describe '.get_metric_png_url' do
+    it 'checks if it gets png image url' do
```
Comment (szmo) [inline rails_app/spec/adapters/graphite_adapter_spec.rb]: returns url to png image

File: rails_app/spec/adapters/graphite_adapter_spec.rb [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+      expect(@graphite).to receive(:get_metric)
+        .with('mock.metric', 'json', time_from: nil, time_to: nil)
+        .and_return(mock_values)
+
+      expect(@graphite.get_metric_json('mock.metric')).to eql(mock_values)
+    end
+  end
+
+  describe '.get_metric_png' do
+    it 'checks if getting png image from graphite' do
```
Comment (szmo) [inline rails_app/spec/adapters/graphite_adapter_spec.rb]: returns png image

File: rails_app/spec/adapters/graphite_adapter_spec.rb [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+      expect(@graphite).to receive(:get_metric_url)
+        .with('mock.metric', 'mock-type', time_from: nil, time_to: nil)
+        .and_return(URI.parse(mock_url))
+
+      expect(@graphite.get_metric('mock.metric', 'mock-type')).to eql([[nil, 1], [1, 2], [2, 3]])
+    end
+  end
+
+  describe '.get_metric_json' do
+    it 'checks if getting json metric' do
```
Comment (szmo) [inline rails_app/spec/adapters/graphite_adapter_spec.rb]: returns metrics in json

File: rails_app/spec/adapters/graphite_adapter_spec.rb [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+        'mock.metric', 'mock-type', time_from: '-2d', time_to: '-1d'
+      ).to_s).to include 'to=-1d'
+      expect(@graphite.get_metric_url(
+        'mock.metric', 'mock-type', time_from: '-2d', time_to: '-1d'
+      ).to_s).to include 'from=-2d'
+    end
+  end
+
+  describe '.get_metric' do
+    it 'fetches metric from graphite and checks json output being casted to hash' do
```
Comment (szmo) [inline rails_app/spec/adapters/graphite_adapter_spec.rb:93]: returns metrics hash

File: rails_app/spec/adapters/graphite_adapter_spec.rb [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+      @graphite.send_raw({ 'mock.metric' => 123 }, { time: 999 })
+    end
+
+    it 'sends raw metric to graphite with metric being hash with value' do
+      expect(api_mock).to receive(:metrics).with('mock.metric' => 123)
+      @graphite.send_raw('mock.metric' => 123)
+    end
+  end
+
+  describe '.send_deal_price' do
```
Comment (szmo) [inline rails_app/spec/adapters/graphite_adapter_spec.rb:45]: its cool sample

File: rails_app/spec/adapters/graphite_adapter_spec.rb [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+  describe '.initialize' do
+    it 'instantiate GraphiteAdapter object and check if default values were loaded' do
+      expect(@graphite.instance_variable_get('@graphite_host')).to eql('localhost')
+      expect(@graphite.instance_variable_get('@default_period')).to eql('-14d')
+      expect(@graphite.instance_variable_get('@api')).to eql(api_mock)
+    end
+  end
+
+  describe '.send_raw' do
```
Comment (szmo) [inline rails_app/spec/adapters/graphite_adapter_spec.rb:23]: its ok

File: rails_app/spec/adapters/graphite_adapter_spec.rb [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+    expect(GraphiteAPI).to receive(:new)
+      .with(graphite: 'localhost:2003', prefix: 'price_monitor_internal')
+      .and_return(api_mock)
+    @graphite = GraphiteAdapter.new
+  end
+
+  let(:api_mock) { double :api_mock }
+
+  describe '.initialize' do
+    it 'instantiate GraphiteAdapter object and check if default values were loaded' do
```
Comment (szmo) [inline rails_app/spec/adapters/graphite_adapter_spec.rb:16]: 'instantiate GraphiteAdapter object' i think its enought

File: rails_app/spec/adapters/graphite_adapter_spec.rb [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+      allow(Time).to receive(:now).and_return(123)
+      expect(@graphite).to receive(:send_raw).with({
+        'mock-user.awesome_product.price'      => 777,
+        'mock-user.awesome_product.check_time' => 123,
+      }, { time: 778 })
+      @graphite.send_deal_price('awesome_product', 777, 778, 'mock-user')
+    end
+  end
+
+  describe '.get_metric_url' do
```
Comment (szmo) [inline rails_app/spec/adapters/graphite_adapter_spec.rb:74]: here to rework

**Review by szmo — CHANGES_REQUESTED** [review]

File: rails_app/spec/adapters/graphite_adapter_spec.rb (hunk 9 above)
Comment (evemorgen) [reply rails_app/spec/adapters/graphite_adapter_spec.rb:16]: Explicit is better than implicit.

File: rails_app/spec/adapters/graphite_adapter_spec.rb (hunk 8 above)
Comment (evemorgen) [reply rails_app/spec/adapters/graphite_adapter_spec.rb:23]: ?

File: rails_app/spec/adapters/graphite_adapter_spec.rb (hunk 7 above)
Comment (evemorgen) [reply rails_app/spec/adapters/graphite_adapter_spec.rb:45]: ?

File: rails_app/spec/adapters/graphite_adapter_spec.rb (hunk 10 above)
Comment (evemorgen) [reply rails_app/spec/adapters/graphite_adapter_spec.rb:74]: ?

File: rails_app/spec/adapters/graphite_adapter_spec.rb (hunk 6 above)
Comment (evemorgen) [reply rails_app/spec/adapters/graphite_adapter_spec.rb:93]: Explicit is better than implicit.

File: rails_app/spec/adapters/graphite_adapter_spec.rb (hunk 5 above)
Comment (evemorgen) [reply rails_app/spec/adapters/graphite_adapter_spec.rb]: ok

File: rails_app/spec/adapters/graphite_adapter_spec.rb (hunk 4 above)
Comment (evemorgen) [reply rails_app/spec/adapters/graphite_adapter_spec.rb]: ok

File: rails_app/spec/adapters/graphite_adapter_spec.rb (hunk 3 above)
Comment (evemorgen) [reply rails_app/spec/adapters/graphite_adapter_spec.rb]: ok

**Review by szmo — APPROVED** [review]

**Pull request merged**